Description: updating makefile
 updating makefile to make it more compatible with packaging & crossbuilding
Author: Nicolas HOUDELOT <nicolas@demosdebs.org>
Last-Update: 2024-03-10
Applied-Upstream: no
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Makefile
+++ b/Makefile
@@ -9,27 +9,29 @@
 LIBS = -lpthread -lm
 SDR_OBJ = cpu.o sdr.o fifo.o sdr_ifile.o dsp/helpers/tables.o
 
+PKG_CONFIG:=pkg-config
+
 # Try to autodetect available libraries via pkg-config if no explicit setting was used
-PKGCONFIG=$(shell pkg-config --version >/dev/null 2>&1 && echo "yes" || echo "no")
+PKGCONFIG=$(shell $(PKG_CONFIG) --version >/dev/null 2>&1 && echo "yes" || echo "no")
 ifeq ($(PKGCONFIG), yes)
   ifndef RTLSDR
     ifdef RTLSDR_PREFIX
       RTLSDR := yes
     else
-      RTLSDR := $(shell pkg-config --exists librtlsdr && echo "yes" || echo "no")
+      RTLSDR := $(shell $(PKG_CONFIG) --exists librtlsdr && echo "yes" || echo "no")
     endif
   endif
 
   ifndef BLADERF
-    BLADERF := $(shell pkg-config --exists libbladeRF && echo "yes" || echo "no")
+    BLADERF := $(shell $(PKG_CONFIG) --exists libbladeRF && echo "yes" || echo "no")
   endif
 
   ifndef HACKRF
-    HACKRF := $(shell pkg-config --exists libhackrf && echo "yes" || echo "no")
+    HACKRF := $(shell $(PKG_CONFIG) --exists libhackrf && echo "yes" || echo "no")
   endif
 
   ifndef LIMESDR
-    LIMESDR := $(shell pkg-config --exists LimeSuite && echo "yes" || echo "no")
+    LIMESDR := $(shell $(PKG_CONFIG) --exists LimeSuite && echo "yes" || echo "no")
   endif
 
   ifndef SOAPYSDR
@@ -107,6 +109,8 @@
 RTLSDR ?= yes
 BLADERF ?= yes
 
+CC:=gcc
+
 ifeq ($(RTLSDR), yes)
   SDR_OBJ += sdr_rtlsdr.o
   DUMP1090_CPPFLAGS += -DENABLE_RTLSDR
@@ -123,18 +127,18 @@
 
     # FreeBSD's librtlsdr.pc includes -std=gnu89 in cflags
     # some linux librtlsdr packages return a bare -I/ with no path in --cflags
-    RTLSDR_CFLAGS := $(shell pkg-config --cflags librtlsdr)
+    RTLSDR_CFLAGS := $(shell $(PKG_CONFIG) --cflags librtlsdr)
     RTLSDR_CFLAGS := $(filter-out -std=%,$(RTLSDR_CFLAGS))
     RTLSDR_CFLAGS := $(filter-out -I/,$(RTLSDR_CFLAGS))
     DUMP1090_CFLAGS += $(RTLSDR_CFLAGS)
 
     # some linux librtlsdr packages return a bare -L with no path in --libs
     # which horribly confuses things because it eats the next option on the command line
-    RTLSDR_LFLAGS := $(shell pkg-config --libs-only-L librtlsdr)
+    RTLSDR_LFLAGS := $(shell $(PKG_CONFIG) --libs-only-L librtlsdr)
     ifeq ($(RTLSDR_LFLAGS),-L)
-      LIBS_SDR += $(shell pkg-config --libs-only-l --libs-only-other librtlsdr)
+      LIBS_SDR += $(shell $(PKG_CONFIG) --libs-only-l --libs-only-other librtlsdr)
     else
-      LIBS_SDR += $(shell pkg-config --libs librtlsdr)
+      LIBS_SDR += $(shell $(PKG_CONFIG) --libs librtlsdr)
     endif
   endif
 endif
@@ -142,22 +146,22 @@
 ifeq ($(BLADERF), yes)
   SDR_OBJ += sdr_bladerf.o
   DUMP1090_CPPFLAGS += -DENABLE_BLADERF
-  DUMP1090_CFLAGS += $(shell pkg-config --cflags libbladeRF)
-  LIBS_SDR += $(shell pkg-config --libs libbladeRF)
+  DUMP1090_CFLAGS += $(shell $(PKG_CONFIG) --cflags libbladeRF)
+  LIBS_SDR += $(shell $(PKG_CONFIG) --libs libbladeRF)
 endif
 
 ifeq ($(HACKRF), yes)
   SDR_OBJ += sdr_hackrf.o
   DUMP1090_CPPFLAGS += -DENABLE_HACKRF
-  DUMP1090_CFLAGS += $(shell pkg-config --cflags libhackrf)
-  LIBS_SDR += $(shell pkg-config --libs libhackrf)
+  DUMP1090_CFLAGS += $(shell $(PKG_CONFIG) --cflags libhackrf)
+  LIBS_SDR += $(shell $(PKG_CONFIG) --libs libhackrf)
 endif
 
 ifeq ($(LIMESDR), yes)
   SDR_OBJ += sdr_limesdr.o
   DUMP1090_CPPFLAGS += -DENABLE_LIMESDR
-  DUMP1090_CFLAGS += $(shell pkg-config --cflags LimeSuite)
-  LIBS_SDR += $(shell pkg-config --libs LimeSuite)
+  DUMP1090_CFLAGS += $(shell $(PKG_CONFIG) --cflags LimeSuite)
+  LIBS_SDR += $(shell $(PKG_CONFIG) --libs LimeSuite)
 endif
 
 ifeq ($(SOAPYSDR), yes)
